src/devtools/client/debugger/src/components/ProjectSearch.js

```diff
@@ -29,6 +29,37 @@ import AccessibleImage from "./shared/AccessibleImage";
 
 import { PluralForm } from "devtools/shared/plural-form";
 import { trackEvent } from "ui/utils/telemetry";
+import { ThreadFront } from "protocol/thread";
+import { groupBy } from "lodash";
+import { isThirdParty } from "../utils/source";
+
+const formatMatches = (matches, sourcesById) => {
+  const resultsBySource = groupBy(matches, res => res.location.sourceId);
+  return Object.entries(resultsBySource).map(([sourceId, matches]) => {
+    return {
+      type: "RESULT",
+      sourceId,
+      filepath: sourcesById[sourceId]?.url,
+      matches: matches.map(match => {
+        // We have to do this array dance to navigate the string in unicode "code points"
+        // because `colunm` is calculated using "code points" as opposed to JS strings
+        // which use "code units". It makes a difference in string with fun unicode characters.
+        const matchStr = [...match.context]
+          .slice(match.contextStart.column, match.contextEnd.column)
+          .join("");
+        return {
```

Comment:
haha I see.

Maybe we want to create a new helper that will make this more obvious and also let us write some unit tests down the line.

`unicodeSlice(match.context, match.contextStart.column, match.contextEnd.column)`

Reply:
@jasonLaster yeah I like that idea. I'm going merge to unblock @jaril and then follow up with a helper.